src/eredis_cluster_manager.erl

```diff
@@ -0,0 +1,415 @@
+-module(eredis_cluster_manager).
+-behaviour(gen_server).
+
+%% API.
+-export([start_link/1]).
+-export([remap_cluster/2]).
+-export([get_eredis_pid/2]).
+-export([qp/2]).
+-export([q/2]).
+
+%% gen_server.
+-export([init/1]).
+-export([handle_call/3]).
+-export([handle_cast/2]).
+-export([handle_info/2]).
+-export([terminate/2]).
+-export([code_change/3]).
+
+%% Type definition.
+-include("eredis_cluster.hrl").
+-record(state, {
+    init_nodes :: [#node{}],
+    slots :: tuple(), %% Tuple with all the Redis slots. Each element is index into slots_maps.
+    slots_maps :: tuple(), %% Tuple with all the Redis nodes and Redis slots mapped on them.
+    version :: integer()  %% Used to avoid unnecessary refresh of Redis slots.
+}).
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% API
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+-spec start_link([]) -> {ok, pid()}.
+start_link({ModName, Args}) ->
+    gen_server:start_link({local, ModName}, ?MODULE, [Args], []).
+
+%% Looks up eredis Pid for Slot.
+-spec get_eredis_pid(Cluster::any(), Slot::integer()) -> 
+    {Pid::pid() | undefined, Version::integer()}.
+get_eredis_pid(Cluster, Slot) ->
+    {ok, Result} = gen_server:call(Cluster, {get_pid_by_slot, Slot}),
+    Result.
+
+
+%% Remaps various redis slots.
+-spec remap_cluster(Cluster::any(), Version::integer()) -> {Version::integer()}.
+remap_cluster(Cluster, Version) ->
```

Comment:
I don't understand this API.. why should the caller mention a version here?
Caller should only ask to remap the cluster if it encounters an error.

Reply:
Using version allows the code to ignore stale requests. I intend to keep this API for internal use. Only redis queries should ideally be used by the caller. Please let me know your comments.

Reply:
yea, I understand. I did not fully see this back then. Thank you vipin!